### ISSUE TITLE
Rating with half star has inadequate contrast

### ISSUE THREAD

**Issue by ecraig12345** [microsoft/fluentui#18178]
### Environment Information

- **Package version(s)**: tested on `@fluentui/react` latest, also applies to 7
- **Browser and OS versions**: any (colors are hardcoded)

### Describe the issue:

When the Rating component has a half-star rating selected, the contrast ratio between the selected and non-selected parts of the half star is only 2:1.

Checked part is #323130 (`palette.neutralPrimary`) and unchecked part is #605E5C (`palette.neutralSecondary`).

### Please provide a reproduction of the issue in a codepen:

https://codepen.io/ecraig12345/pen/Vwpaaqm?editors=1010

#### Actual behavior:

2:1 contrast in the half-checked star

![image](https://user-images.githubusercontent.com/5864305/118177972-410ba980-b3e8-11eb-9093-678c65caec5e.png)

#### Expected behavior:

I assume the checked and unchecked parts of the star should have at least the standard 4.5:1 contrast ratio (don't see any reason why this would be exempt).

The half-star state isn't shown in the figma toolkit, so ideally we should check with design about what they want. One possibility would be making the unselected part of the star should have a white (body) background, maybe by changing this line to use `IRatingProps.unselectedIcon` if it's a half-filled star.
https://github.com/microsoft/fluentui/blob/2e6d5cd387a0bc07077c5cb4b7f6eef08db8762f/packages/react/src/components/Rating/Rating.base.tsx#L13
